### PR TITLE
[java] UseArraysAsList: Description added

### PR DESCRIPTION
**Pull request by tweimer**
I added a note about UseArraysAsList, which should IMHO be mentioned, as Arrays.asList(..) is not exactly equivalent to the original code. I hope I made my point clear.

Feel free to rephrase it if you like (hopefully I didn't make many typos).

### CODE REVIEW

**Comment by jsotuyod** [comment]
@tweimer thanks for the PR! We will look into adding this for 6.1.0, as 6.0.1 is a hotfixing release.

File: pmd-java/src/main/resources/category/java/performance.xml [hunk 1]
```diff
@@ -790,6 +790,11 @@ public class SimpleTest extends TestCase {
         <description>
 The java.util.Arrays class has a "asList" method that should be used when you want to create a new List from
 an array of objects. It is faster than executing a loop to copy all the elements of the array one by one.
+
+Note that the result of Arrays.asList() is backed by the specified array,
+changes in the returned list will result in the array to be modified.
+For that reason, it is not possible to add new elements to the returned list of Arrays.asList() (UnsupportedOperationException).
+You must use new ArrayList<>(Arrays.asList(...)) if that is inconvenient for you (e.g. because of concurrent access).
```
Comment (jsotuyod) [inline pmd-java/src/main/resources/category/java/performance.xml]: to be able to user `<` and `>`, you should wrap the whole `<description>` contents under a CDATA

Comment (tweimer) [reply pmd-java/src/main/resources/category/java/performance.xml]: Thabks, I will fix that.

Comment (tweimer) [reply pmd-java/src/main/resources/category/java/performance.xml]: Done.

**Comment by jsotuyod** [comment]
Merged! Thanks for the contribution and patience!

**Pull request merged**